kie-server-parent/kie-server-tests/kie-server-integ-tests-optaplanner/src/test/java/org/kie/server/integrationtests/optaplanner/OptaplannerIntegrationTest.java

```diff
@@ -0,0 +1,326 @@
+        assertSuccess(createSolverResponse);
+
+        createSolverResponse = solverClient.createSolver( CONTAINER_1_ID, SOLVER_1_ID, instance );
+        ServiceResponse.ResponseType type = createSolverResponse.getType();
+        assertEquals( "Expected FAILURE response, but got " + type + "!", ServiceResponse.ResponseType.FAILURE, type );
+    }
+
+    @Ignore @Test(expected = KieServicesException.class)
+    public void testDisposeNotExistingSolver() throws Exception {
+        solverClient.disposeSolver( CONTAINER_1_ID, SOLVER_1_ID );
+    }
+
+    @Test
+    public void testGetSolverState() throws Exception {
+        SolverInstance instance = new SolverInstance();
+        instance.setSolverConfigFile( SOLVER_1_CONFIG );
+        assertSuccess( client.createContainer( CONTAINER_1_ID, new KieContainerResource( CONTAINER_1_ID, kjar1 ) ) );
+        assertSuccess( solverClient.createSolver( CONTAINER_1_ID, SOLVER_1_ID, instance ) );
+
+        ServiceResponse<SolverInstance> solverState = solverClient.getSolverState(CONTAINER_1_ID, SOLVER_1_ID);
+        assertSuccess(solverState);
+
+        SolverInstance returnedInstance = solverState.getResult();
+        assertEquals(CONTAINER_1_ID, returnedInstance.getContainerId());
+        assertEquals( SOLVER_1_CONFIG, returnedInstance.getSolverConfigFile() );
+        assertEquals( SOLVER_1_ID, returnedInstance.getSolverId() );
+        assertEquals( SolverInstance.getSolverInstanceKey( CONTAINER_1_ID, SOLVER_1_ID ), returnedInstance.getSolverInstanceKey());
+        assertEquals( SolverInstance.SolverStatus.NOT_SOLVING, returnedInstance.getStatus());
+        assertNull( returnedInstance.getScore() );
+    }
+
+    @Test
+    public void testGetNotExistingSolverState() throws Exception {
+        assertSuccess( client.createContainer( CONTAINER_1_ID, new KieContainerResource( CONTAINER_1_ID, kjar1 ) ) );
+
+        try {
+            solverClient.getSolverState(CONTAINER_1_ID, SOLVER_1_ID);
+        } catch (KieServicesException e) {
+            assertResultContainsStringRegex(e.getMessage(), ".*Solver.*not found in container.*");
+        }
+    }
+
+    @Test
+    public void testExecuteSolver() throws Exception {
+        SolverInstance instance = new SolverInstance();
+        instance.setSolverConfigFile( SOLVER_1_CONFIG );
+        assertSuccess( client.createContainer( CONTAINER_1_ID, new KieContainerResource( CONTAINER_1_ID, kjar1 ) ) );
+
+        ServiceResponse<SolverInstance> response = solverClient.createSolver( CONTAINER_1_ID, SOLVER_1_ID, instance );
+        assertSuccess( response );
+        assertEquals( SolverInstance.SolverStatus.NOT_SOLVING, response.getResult().getStatus() );
+
+        // the following status starts the solver
+        instance.setStatus( SolverInstance.SolverStatus.SOLVING );
+        instance.setPlanningProblem( loadPlanningProblem( 5, 10 ) );
+        response = solverClient.updateSolverState( CONTAINER_1_ID, SOLVER_1_ID, instance );
```

Comment:
What about creating own separate methods in client for starting solver and terminating it? It would be more intuitive and user friendly from my point of view. Also it would show what parameters are mandatory for each action.